plugins/source/hackernews/client/testing.go

```diff
@@ -7,52 +7,72 @@ import (
 	"time"
 
 	"github.com/cloudquery/cloudquery/plugins/source/hackernews/client/services"
-	"github.com/cloudquery/plugin-pb-go/specs"
-	"github.com/cloudquery/plugin-sdk/v3/backend"
-	"github.com/cloudquery/plugin-sdk/v3/plugins/source"
-	"github.com/cloudquery/plugin-sdk/v3/schema"
-	"github.com/golang/mock/gomock"
+	"github.com/cloudquery/plugin-sdk/v4/scheduler"
+	"github.com/cloudquery/plugin-sdk/v4/schema"
+	"github.com/cloudquery/plugin-sdk/v4/transformers"
 	"github.com/rs/zerolog"
 )
 
 type TestOptions struct {
-	Backend   backend.Backend
 	StartTime time.Time
 }
 
-func MockTestHelper(t *testing.T, table *schema.Table, builder func(*testing.T, *gomock.Controller) services.HackernewsClient, opts TestOptions) {
-	version := "vDev"
+func MockTestHelper(t *testing.T, table *schema.Table, builder func(*testing.T) services.HackernewsClient, opts TestOptions) {
+	// version := "vDev"
 	table.IgnoreInTests = false
 	t.Helper()
-	ctrl := gomock.NewController(t)
+	//ctrl := gomock.NewController(t)
 	l := zerolog.New(zerolog.NewTestWriter(t)).Output(
 		zerolog.ConsoleWriter{Out: os.Stderr, TimeFormat: time.StampMicro},
 	).Level(zerolog.DebugLevel).With().Timestamp().Logger()
-	newTestExecutionClient := func(ctx context.Context, logger zerolog.Logger, spec specs.Source, _ source.Options) (schema.ClientMeta, error) {
-		startTime := ""
-		if !opts.StartTime.IsZero() {
-			startTime = opts.StartTime.Format(time.RFC3339)
-		}
-		return &Client{
-			logger:     l,
-			Backend:    opts.Backend,
-			HackerNews: builder(t, ctrl),
-			Spec:       Spec{ItemConcurrency: 10, StartTime: startTime},
-		}, nil
+
+	schedulerClient := &Client{
+		logger:     l,
+		HackerNews: builder(t),
+		// Backend:    mockBackend, TODO(v4)
+		maxRetries: 0,
+		backoff:    1 * time.Millisecond,
+	}
+	tables := schema.Tables{table}
+	if err := transformers.TransformTables(tables); err != nil {
+		t.Fatal(err)
+	}
+	sc := scheduler.NewScheduler(schedulerClient, scheduler.WithLogger(l))
```

Comment:
unrelated to this PR but seems we will need to move the `schedulerClient` to `scheduler.Sync` because backend is a per `sync` option